### ISSUE TITLE
Flattening a folded group causes children to disappear

### ISSUE THREAD

**Issue by jathek** [mbnuqw/sidebery#1158]
## Steps to reproduce
- Create a tree
- Fold the tree
- Flatten the tree
- Children disappear

https://github.com/mbnuqw/sidebery/assets/692914/44b38aae-697a-47c4-9e8d-6e3b63d9ff33

## Expected behavior
Group is unfolded and flattened

## Actual behavior
Children disappear and only parent remains

## Debug info

<details>
<summary><b>Addon data</b></summary>
     
```json
{
  "addonVersion": "5.0.0rc4",
  "firefoxVersion": "117.0",
  "settings": {
    "nativeScrollbars": true,
    "nativeScrollbarsThin": true,
    "nativeScrollbarsLeft": false,
    "selWinScreenshots": true,
    "updateSidebarTitle": true,
    "markWindow": true,
    "markWindowPreface": "​",
    "ctxMenuNative": false,
    "ctxMenuRenderInact": false,
    "ctxMenuRenderIcons": true,
    "ctxMenuIgnoreContainers": "",
    "navBarLayout": "horizontal",
    "navBarInline": true,
    "navBarSide": "left",
    "hideAddBtn": true,
    "hideSettingsBtn": false,
    "navBtnCount": true,
    "hideEmptyPanels": false,
    "hideDiscardedTabPanels": false,
    "navActTabsPanelLeftClickAction": "scroll",
    "navActBookmarksPanelLeftClickAction": "none",
    "navTabsPanelMidClickAction": "none",
    "navBookmarksPanelMidClickAction": "none",
    "navSwitchPanelsWheel": true,
    "subPanelRecentlyClosedBar": true,
    "subPanelBookmarks": true,
    "subPanelHistory": true,
    "groupLayout": "grid",
    "skipEmptyPanels": true,
    "dndTabAct": false,
    "dndTabActDelay": 750,
    "dndTabActMod": "none",
    "dndExp": "pointer",
    "dndExpDelay": 0,
    "dndExpMod": "none",
    "dndOutside": "data",
    "dndActTabFromLink": true,
    "dndActSearchTab": true,
    "dndMoveTabs": false,
    "dndMoveBookmarks": false,
    "searchBarMode": "dynamic",
    "searchBookmarksShortcut": "",
    "searchHistoryShortcut": "",
    "warnOnMultiTabClose": "any",
    "activateLastTabOnPanelSwitching": true,
    "activateLastTabOnPanelSwitchingLoadedOnly": true,
    "switchPanelAfterSwitchingTab": "always",
    "tabRmBtn": "none",
    "activateAfterClosing": "next",
    "activateAfterClosingStayInPanel": true,
    "activateAfterClosingGlobal": false,
    "activateAfterClosingNoFolded": true,
    "activateAfterClosingNoDiscarded": true,
    "askNewBookmarkPlace": true,
    "tabsRmUndoNote": true,
    "tabsUnreadMark": true,
    "tabsUpdateMark": "all",
    "tabsUpdateMarkFirst": true,
    "tabsReloadLimit": 5,
    "tabsReloadLimitNotif": true,
    "showNewTabBtns": true,
    "newTabBarPosition": "bottom",
    "tabsPanelSwitchActMove": true,
    "tabsPanelSwitchActMoveAuto": true,
    "tabsUrlInTooltip": "full",
    "newTabCtxReopen": false,
    "moveNewTabPin": "end",
    "moveNewTabParent": "last_child",
    "moveNewTabParentActPanel": false,
    "moveNewTab": "end",
    "moveNewTabActivePin": "start",
    "pinnedTabsPosition": "panel",
    "pinnedTabsList": false,
    "pinnedAutoGroup": true,
    "pinnedNoUnload": false,
    "tabsTree": true,
    "groupOnOpen": true,
    "tabsTreeLimit": "none",
    "autoFoldTabs": false,
    "autoFoldTabsExcept": 5,
    "autoExpandTabs": true,
    "autoExpandTabsOnNew": false,
    "rmChildTabs": "folded",
    "tabsChildCount": true,
    "tabsLvlDots": true,
    "discardFolded": false,
    "discardFoldedDelay": 0,
    "discardFoldedDelayUnit": "sec",
    "tabsTreeBookmarks": true,
    "treeRmOutdent": "first_child",
    "ignoreFoldedParent": false,
    "colorizeTabs": false,
    "colorizeTabsSrc": "domain",
    "colorizeTabsBranches": false,
    "colorizeTabsBranchesSrc": "url",
    "inheritCustomColor": true,
    "hideInact": true,
    "hideFoldedTabs": true,
    "hideFoldedParent": "none",
    "nativeHighlight": true,
    "warnOnMultiBookmarkDelete": "any",
    "autoCloseBookmarks": false,
    "autoRemoveOther": false,
    "highlightOpenBookmarks": true,
    "activateOpenBookmarkTab": true,
    "showBookmarkLen": true,
    "bookmarksRmUndoNote": true,
    "loadBookmarksOnDemand": true,
    "pinOpenedBookmarksFolder": true,
    "oldBookmarksAfterSave": "ask",
    "loadHistoryOnDemand": true,
    "fontSize": "m",
    "animations": true,
    "animationSpeed": "fast",
    "theme": "proton",
    "density": "default",
    "colorScheme": "ff",
    "sidebarCSS": true,
    "groupCSS": false,
    "snapNotify": true,
    "snapExcludePrivate": false,
    "snapInterval": 240,
    "snapIntervalUnit": "min",
    "snapLimit": 14,
    "snapLimitUnit": "day",
    "snapAutoExport": false,
    "snapAutoExportType": "json",
    "snapAutoExportPath": "Sidebery/snapshot-%Y.%M.%D-%h.%m.%s",
    "snapMdFullTree": false,
    "hScrollAction": "none",
    "navSwitchPanelsDelay": 128,
    "scrollThroughTabs": "none",
    "scrollThroughVisibleTabs": false,
    "scrollThroughTabsSkipDiscarded": false,
    "scrollThroughTabsExceptOverflow": true,
    "scrollThroughTabsCyclic": false,
    "scrollThroughTabsScrollArea": 0,
    "autoMenuMultiSel": true,
    "multipleMiddleClose": false,
    "longClickDelay": 500,
    "wheelThreshold": false,
    "wheelThresholdX": 10,
    "wheelThresholdY": 60,
    "tabDoubleClick": "reload",
    "tabsSecondClickActPrev": false,
    "tabsSecondClickActPrevPanelOnly": false,
    "shiftSelAct": true,
    "activateOnMouseUp": true,
    "tabLongLeftClick": "none",
    "tabLongRightClick": "edit_title",
    "tabCloseMiddleClick": "close",
    "tabsPanelLeftClickAction": "none",
    "tabsPanelDoubleClickAction": "tab",
    "tabsPanelRightClickAction": "menu",
    "tabsPanelMiddleClickAction": "undo",
    "newTabMiddleClickAction": "new_child",
    "bookmarksLeftClickAction": "open_in_act",
    "bookmarksLeftClickActivate": false,
    "bookmarksLeftClickPos": "default",
    "bookmarksMidClickAction": "open_in_new",
    "bookmarksMidClickActivate": false,
    "bookmarksMidClickRemove": false,
    "bookmarksMidClickPos": "default",
    "syncName": "DeeperThought",
    "syncSaveSettings": true,
    "syncSaveCtxMenu": true,
    "syncSaveStyles": true,
    "syncSaveKeybindings": false,
    "selectActiveTabFirst": true
  },
  "permissions": {
    "allUrls": true,
    "webRequest": true,
    "webRequestBlocking": true,
    "proxy": true,
    "tabHide": true,
    "clipboardWrite": true,
    "history": true,
    "bookmarks": true,
    "downloads": false
  },
  "storage": {
    "size": "4.03 mb",
    "props": {
      "bookmarksRecentFolders": "76 b",
      "containers": "4.22 kb",
      "contextMenu": "1.3 kb",
      "disabledKeybindings": "24 b",
      "expandedBookmarkFolders": "175 b",
      "favDomains": "492 kb",
      "favHashes": "21.4 kb",
      "favicons": "1.54 mb",
      "groupCSS": "2 b",
      "lastSnapTime": "13 b",
      "panelIndex": "1 b",
      "prevTabsDataCache": "115 kb",
      "profileID": "14 b",
      "settings": "4.38 kb",
      "sidebar": "3.11 kb",
      "sidebarCSS": "282 b",
      "snapshots": "1.7 mb",
      "tabsDataCache": "166 kb",
      "ver": "10 b"
    }
  },
  "sidebar": {
    "nav": [
      "firefox-default",
      "aYh-zk8WQ8ai",
      "o-DtX9U7s64h",
      "TLGJZqGazkVg",
      "Qf9vV8t_uFI9",
      "WReBwPxZP5Sg",
      "bookmarks",
      "history",
      "create_snapshot",
      "search",
      "settings"
    ],
    "panels": {
      "firefox-default": {
        "type": 2,
        "id": "firefox-default",
        "name": "len: 7",
        "color": "toolbar",
        "iconSVG": "icon_tabs",
        "iconIMGSrc": "",
        "iconIMG": "",
        "lockedPanel": false,
        "skipOnSwitching": false,
        "noEmpty": false,
        "newTabCtx": "none",
        "dropTabCtx": "none",
        "moveTabCtx": "none",
        "moveTabCtxNoChild": true,
        "bookmarksFolderId": -1,
        "newTabBtns": [],
        "moveRules": [],
        "moveExcludedTo": -1,
        "srcPanelConfig": null
      },
      "aYh-zk8WQ8ai": {
        "type": 2,
        "id": "aYh-zk8WQ8ai",
        "name": "len: 5",
        "color": "blue",
        "iconSVG": "icon_edu",
        "iconIMGSrc": "",
        "iconIMG": "",
        "lockedPanel": false,
        "skipOnSwitching": true,
        "noEmpty": false,
        "newTabCtx": "none",
        "dropTabCtx": "none",
        "moveTabCtx": "none",
        "moveTabCtxNoChild": true,
        "bookmarksFolderId": -1,
        "newTabBtns": [],
        "moveRules": [],
        "moveExcludedTo": -1,
        "srcPanelConfig": null
      },
      "Qf9vV8t_uFI9": {
        "type": 2,
        "id": "Qf9vV8t_uFI9",
        "name": "len: 7",
        "color": "turquoise",
        "iconSVG": "icon_code",
        "iconIMGSrc": "",
        "iconIMG": "",
        "lockedPanel": false,
        "skipOnSwitching": false,
        "noEmpty": false,
        "newTabCtx": "none",
        "dropTabCtx": "none",
        "moveTabCtx": "none",
        "moveTabCtxNoChild": true,
        "bookmarksFolderId": "4yldqbvKJbRH",
        "newTabBtns": [],
        "moveRules": [],
        "moveExcludedTo": -1,
        "srcPanelConfig": null
      },
      "WReBwPxZP5Sg": {
        "type": 2,
        "id": "WReBwPxZP5Sg",
        "name": "len: 9",
        "color": "purple",
        "iconSVG": "icon_play",
        "iconIMGSrc": "",
        "iconIMG": "",
        "lockedPanel": false,
        "skipOnSwitching": true,
        "noEmpty": false,
        "newTabCtx": "none",
        "dropTabCtx": "none",
        "moveTabCtx": "none",
        "moveTabCtxNoChild": true,
        "bookmarksFolderId": -1,
        "newTabBtns": [],
        "moveRules": [],
        "moveExcludedTo": -1,
        "srcPanelConfig": null
      },
      "bookmarks": {
        "type": 1,
        "id": "bookmarks",
        "name": "len: 9",
        "iconSVG": "icon_bookmarks",
        "iconIMGSrc": "",
        "iconIMG": "",
        "color": "toolbar",
        "lockedPanel": false,
        "tempMode": false,
        "skipOnSwitching": false,
        "rootId": "root________",
        "viewMode": "tree",
        "autoConvert": false,
        "srcPanelConfig": null
      },
      "history": {
        "type": 4,
        "id": "history",
        "name": "len: 7",
        "color": "toolbar",
        "iconSVG": "icon_clock",
        "tempMode": false,
        "lockedPanel": false,
        "skipOnSwitching": false,
        "viewMode": "history"
      },
      "Oa4d4I6dRg4i": {
        "type": 2,
        "id": "Oa4d4I6dRg4i",
        "name": "len: 7",
        "color": "yellow",
        "iconSVG": "icon_flask",
        "iconIMGSrc": "",
        "iconIMG": "",
        "lockedPanel": true,
        "skipOnSwitching": true,
        "noEmpty": false,
        "newTabCtx": "none",
        "dropTabCtx": "none",
        "moveRules": [],
        "moveExcludedTo": -1,
        "bookmarksFolderId": -1,
        "newTabBtns": [],
        "srcPanelConfig": null
      }
    }
  },
  "containers": [
    {
      "id": "firefox-container-8",
      "name": "6",
      "icon": "...",
      "color": "turquoise",
      "proxified": false,
      "proxy": {
        "type": "direct"
      },
      "userAgentActive": false,
      "userAgent": "",
      "cookieStoreId": "firefox-container-8",
      "colorCode": "#37adff",
      "reopenRulesActive": false,
      "reopenRules": []
    },
    {
      "id": "firefox-container-29",
      "name": "9",
      "icon": "...",
      "color": "pink",
      "proxified": true,
      "proxy": {
        "type": "http"
      },
      "userAgentActive": false,
      "userAgent": "",
      "cookieStoreId": "firefox-container-29",
      "colorCode": "#37adff",
      "reopenRulesActive": true,
      "reopenRules": []
    },
    {
      "id": "firefox-container-30",
      "name": "19",
      "icon": "...",
      "color": "green",
      "proxified": "valid",
      "proxy": {
        "type": "http"
      },
      "userAgentActive": true,
      "userAgent": "102",
      "cookieStoreId": "firefox-container-30",
      "colorCode": "#37adff",
      "reopenRulesActive": true,
      "reopenRules": []
    },
    {
      "id": "firefox-container-31",
      "name": "18",
      "icon": "...",
      "color": "green",
      "proxified": true,
      "proxy": {
        "type": "http"
      },
      "userAgentActive": true,
      "userAgent": "102",
      "cookieStoreId": "firefox-container-31",
      "colorCode": "#37adff",
      "reopenRulesActive": false,
      "reopenRules": []
    },
    {
      "id": "firefox-container-37",
      "name": "25",
      "icon": "...",
      "color": "green",
      "proxified": "valid",
      "proxy": {
        "type": "http"
      },
      "userAgentActive": true,
      "userAgent": "102",
      "cookieStoreId": "firefox-container-37",
      "colorCode": "#37adff",
      "reopenRulesActive": false,
      "reopenRules": []
    },
    {
      "id": "firefox-container-41",
      "name": "18",
      "icon": "...",
      "color": "green",
      "proxified": "valid",
      "proxy": {
        "type": "http"
      },
      "userAgentActive": true,
      "userAgent": "102",
      "cookieStoreId": "firefox-container-41",
      "colorCode": "#37adff",
      "reopenRulesActive": false,
      "reopenRules": []
    },
    {
      "id": "firefox-container-42",
      "name": "18",
      "icon": "...",
      "color": "green",
      "proxified": "valid",
      "proxy": {
        "type": "http"
      },
      "userAgentActive": true,
      "userAgent": "102",
      "cookieStoreId": "firefox-container-42",
      "colorCode": "#37adff",
      "reopenRulesActive": true,
      "reopenRules": []
    },
    {
      "id": "firefox-container-43",
      "cookieStoreId": "firefox-container-43",
      "name": "9",
      "icon": "...",
      "color": "green",
      "colorCode": "#37adff",
      "proxified": true,
      "proxy": {
        "type": "http"
      },
      "userAgentActive": true,
      "userAgent": "102",
      "reopenRulesActive": false,
      "reopenRules": []
    }
  ],
  "sidebarCSSLen": "269",
  "windows": [
    {
      "state": "maximized",
      "incognito": false,
      "tabsCount": 1106
    }
  ],
  "bookmarks": {
    "bookmarksCount": 1278,
    "foldersCount": 56,
    "separatorsCount": 0,
    "maxDepth": 10
  }
}
```
</details>
<b>Logs</b>

```
no logs happen 🙁
```

**Comment by albino1** [comment]
Interestingly it seems like they're still visible if you have native tabs enabled, they're just hidden in Sidebery.

**Comment by mbnuqw** [comment]
Resolved in v5.0.0
